### PR TITLE
Create GithubApp and use it to Fetch views and Contributors

### PR DESCRIPTION
**Pull request by MaggieKimani1**
## Description

- Create a GithubApp using octokit
- Install it within MicrosoftGraph organization's account
- Use the installation token to fetch views and contributors fields
- Populate said data on UI
### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshot
image

### CODE REVIEW

File: SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs [hunk 1]
```diff
@@ -141,23 +143,31 @@ public async Task<List<Node>> GetRepositories(string name, string endCursor = nu
...
+
+            // Create a new GitHubClient using the installation token as authentication
+            var installationClient = new GitHubClient(new ProductHeaderValue(_config.GetValue<string>("product")))
+            {
+                Credentials = new Credentials(token)
+            };
+            return installationClient;
+        }
+
+        internal async Task<Dictionary<string, string>> FetchContributors(string repoName)
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs:164]: The http client should be registered at the Startup.cs and then set the messagehandler.

File: SamplesDashboard/SamplesDashboard/Services/GithubAuthService.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+using Octokit;
+using SamplesDashboard.Helper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SamplesDashboard.Services
+{
+    public class GithubAuthService
```
Comment (andrueastman) [inline SamplesDashboard/SamplesDashboard/Services/GithubAuthService.cs:18]: Lets add XML comments to the functions in this file

File: SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs (hunk 1 above)
Comment (MaggieKimani1) [reply SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs:164]: What do you mean by setting the messagehandler?

Comment (MaggieKimani1) [reply SamplesDashboard/SamplesDashboard/Services/RepositoriesService.cs:164]: Got it!

**Pull request merged**